src/libcore/num/mod.rs

```diff
@@ -1471,3 +1472,73 @@ impl fmt::Display for ParseIntError {
 }
 
 pub use num::dec2flt::ParseFloatError;
+
+
+// Conversion traits for primitive integer types
+// Conversions T -> T are covered by a blanket impl and therefore excluded
+macro_rules! impl_from {
+    ($Small: ty, $Large: ty) => {
+        impl From<$Small> for $Large {
+            #[stable(feature = "lossless_int_conv", since = "1.5.0")]
+            #[inline]
+            fn from(small: $Small) -> $Large {
+                small as $Large
+            }
+        }
+    }
+}
+
+// Unsigned -> Unsigned
+impl_from! { u8, u16 }
+impl_from! { u8, u32 }
+impl_from! { u8, u64 }
+impl_from! { u8, usize }
+impl_from! { u16, u32 }
+impl_from! { u16, u64 }
+impl_from! { u16, usize }
+impl_from! { u32, u64 }
+#[cfg(any(target_pointer_width = "64", target_pointer_width = "32"))]
+impl_from! { u32, usize }
+#[cfg(target_pointer_width = "64")]
```

Comment:
Yeah I would prefer to avoid `usize` and `isize` in these "always working" coercions for now. I'm not actually aware of any types which have some traits implemented on some platforms and not others, and that's a great boon for portability so it'd be nice to keep it that way!